src/components/Program/ProgramCard.js

```diff
@@ -6,32 +6,53 @@ import Dropdown from 'antd/lib/dropdown';
 import { useDispatch } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { programActions } from '../../state/ducks';
+import { useUserRole } from '../../hooks';
 import { pathUtils } from '../../routes';
 import { MenuItemLink } from '../_common';
 // css
 import '../../styles/ProgramCard.css';
 
 export default function ProgramCard(props) {
-  const { program } = props;
+  const { program, push } = props;
   const dispatch = useDispatch();
 
-  const menu = (
-    <Menu>
-      <MenuItemLink
-        to={pathUtils.makeEditProgramPath(program.programid)}
-        key="edit"
-        extra=""
-      >
-        Edit Program
-      </MenuItemLink>
-      <MenuItemLink
+  const { userIsAdmin } = useUserRole();
+
+  const handleMenuClick = e => {
+    if (e.key === 'edit') {
+      dispatch(programActions.setEdit(program.programid));
+      // push(pathUtils.makeEditProgramPath(program.programId));
+      //^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ FIX PUSH ^^^^^^^^^^^^^^^^^^^^^^^^^//
+    } else {
+      dispatch(programActions.deleteProgramThunk(program.programid));
```

Comment:
this bug is actually not your fault — looks like the backend is refusing to delete this program since its sub-courses are referenced in other tables (like modules).

Reply:
sweet thanks!